app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,100 @@
+class InvitationsController < ApplicationController
+  before_action :restrict_access_on_create, only: [ :new, :create ]
+  before_action :restrict_access_on_update, only: [:edit, :update]
+
+  # GET /invitations/new
+  def new
+    @user = User.new
```

Comment:
I think it could be better not to create a user before actual sign up. What if an invited person doesn't respond to an invitation link but is trying to sign up with the same email later?